algoliasearch/client_test.go

```diff
@@ -430,3 +432,75 @@ func TestDnsTimeout(t *testing.T) {
 		t.Fatalf("TestDnsTimeout: Spent %d seconds instead of <5s to perform the 10 retries", int(delta.Seconds()))
 	}
 }
+
+func TestMultiClusterManagement(t *testing.T) {
+	client := initMCMClient(t)
+	var uniqueUserID = fmt.Sprintf("go-MCM-tests-%d", rand.Int())
+
+	t.Log("TestMultiClusterManagement: Listing clusters")
+	clusters, err := client.ListClusters()
+	if err != nil {
+		t.Fatalf("TestMultiClusterManagement: Cannot list Clusters: %s", err)
+	}
+	if len(clusters) <= 1 {
+		t.Log("The app as 1 or less clusters, which is odd for an MCM app")
+	}
+
+	clusterName := clusters[0].ClusterName
+
+	t.Logf("TestMultiClusterManagement: Assigning userID [%s] to cluster [%s]", uniqueUserID, clusterName)
+	_, err = client.AssignUserID(uniqueUserID, clusterName)
+	if err != nil {
+		t.Fatalf("TestMultiClusterManagement: Cannot assign a userIDs: %s", err)
+	}
+
+	t.Logf("TestMultiClusterManagement: Getting userID [%s]", uniqueUserID)
+	var userID UserID
+	for i := 0; i < 10; i++ {
+		userID, err = client.GetUserID(uniqueUserID)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(i) * time.Second)
+	}
+	if userID.ClusterName == "" {
+		t.Fatalf("TestMultiClusterManagement: Cannot get assigned userIDs: %s", err)
+	}
+
+	t.Log("TestMultiClusterManagement: Listing userIDs")
+	users, err := client.ListUserIDs(0, 10)
+	if err != nil {
+		t.Fatalf("TestMultiClusterManagement: Cannot list userIDs: %s", err)
+	}
+	if len(users.UserIDs) <= 1 {
+		t.Fatalf("The app has 1 or less clusters, which is odd considering we just added one")
```

Comment:
Could you prefix the error with `TestMultiClusterManagement: ` (it's really easier for debugging later on, trust me ^^). Same goes for other places where it's missing, if any.